server/src/internalClusterTest/java/org/opensearch/search/backpressure/SearchBackpressureIT.java

```diff
@@ -0,0 +1,268 @@
+
+        CountDownLatch latch = new CountDownLatch(1);
+
+        client().execute(TestTransportAction.ACTION, new TestRequest(RequestType.HIGH_ELAPSED_TIME), new ActionListener<>() {
+            @Override
+            public void onResponse(TestResponse testResponse) {
+                fail("SearchShardTask should have been cancelled");
+                latch.countDown();
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                assertEquals(TaskCancelledException.class, e.getClass());
+                assertTrue(e.getMessage().contains("elapsed time exceeded"));
+                latch.countDown();
+            }
+        });
+
+        latch.await(TIMEOUT.getSeconds(), TimeUnit.SECONDS);
+    }
+
+    public void testSearchShardTaskCancellationWithHighCpu() throws InterruptedException {
+        Settings request = Settings.builder()
+            .put(SearchBackpressureSettings.SETTING_MODE.getKey(), "enforced")
+            .put(CpuUsageTracker.SETTING_CPU_TIME_MILLIS_THRESHOLD.getKey(), 15000)
+            .build();
+        assertAcked(client().admin().cluster().prepareUpdateSettings().setPersistentSettings(request).get());
+
+        CountDownLatch latch = new CountDownLatch(1);
+
+        client().execute(TestTransportAction.ACTION, new TestRequest(RequestType.HIGH_CPU), new ActionListener<>() {
+            @Override
+            public void onResponse(TestResponse testResponse) {
+                fail("SearchShardTask should have been cancelled");
+                latch.countDown();
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                assertEquals(TaskCancelledException.class, e.getClass());
+                assertTrue(e.getMessage().contains("cpu usage exceeded"));
+                latch.countDown();
+            }
+        });
+
+        latch.await(TIMEOUT.getSeconds(), TimeUnit.SECONDS);
+    }
+
+    public void testSearchCancellationWithBackpressureDisabled() throws InterruptedException {
+        Settings request = Settings.builder().put(SearchBackpressureSettings.SETTING_MODE.getKey(), "monitor_only").build();
+        assertAcked(client().admin().cluster().prepareUpdateSettings().setPersistentSettings(request).get());
+
+        CountDownLatch latch = new CountDownLatch(1);
+
+        client().execute(TestTransportAction.ACTION, new TestRequest(RequestType.HIGH_ELAPSED_TIME), new ActionListener<>() {
+            @Override
+            public void onResponse(TestResponse testResponse) {
+                latch.countDown();
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                fail("SearchShardTask shouldn't have cancelled for monitor_only mode");
+                latch.countDown();
+            }
+        });
+
+        latch.await(TIMEOUT.getSeconds(), TimeUnit.SECONDS);
+    }
+
+    enum RequestType {
```

Comment:
Do you plan to add a test for high heap usage as well?

Reply:
Not planning as of now. I will take it up in future.

Reply:
Added high heap usage tests as well.